### PR TITLE
Discovery improvements

### PR DESCRIPTION
**Pull request by jjkester**
Documentation, cleanup, `Instance` improvements, new config discovery

### CODE REVIEW

File: implementation/discovery/api/src/main/java/org/inaetics/dronessimulator/discovery/api/tree/TreeNode.java [hunk 1]
```diff
@@ -4,17 +4,44 @@
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
+/**
+ * Abstract class for a node in a tree.
+ *
+ * @param <K> Type of the keys.
+ * @param <V> Type of the values.
+ * @param <N> Type of the parent and child nodes.
```
Comment (ghost) [inline implementation/discovery/api/src/main/java/org/inaetics/dronessimulator/discovery/api/tree/TreeNode.java]: and current node

File: implementation/discovery/etcd/src/main/java/org/inaetics/dronessimulator/discovery/etcd/DiscoveryStoredEtcdNode.java [hunk 2]
```diff
@@ -9,9 +9,19 @@
 import java.util.Map;
 import java.util.stream.Collectors;
 
+/**
+ * Implementation of a stored node for etcd.
+ *
+ * Maps key/value pairs to distinct etcd nodes.
```
Comment (ghost) [inline implementation/discovery/etcd/src/main/java/org/inaetics/dronessimulator/discovery/etcd/DiscoveryStoredEtcdNode.java]: Squashes a etcd directory node with zero or more values into a single DiscoveryStoredNode.

File: implementation/discovery/etcd/src/main/java/org/inaetics/dronessimulator/discovery/etcd/EtcdChangeHandler.java [hunk 3]
```diff
@@ -54,6 +65,14 @@ public void run() {
         Logger.getLogger(EtcdChangeHandler.class).info("Stopped EtcdChangeHandler!");
     }
 
+    /**
+     * Adds handlers for node events.
```
Comment (ghost) [inline implementation/discovery/etcd/src/main/java/org/inaetics/dronessimulator/discovery/etcd/EtcdChangeHandler.java]: They are added to the root of the tree so they will receive all respective events.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by jjkester** [comment]
Should be fixed now.

**Pull request merged**